js/style/style_batch.js

```diff
@@ -46,11 +48,22 @@ styleBatch.prototype = {
         if (this._style._layers[layer.id] !== undefined) {
             throw new Error('There is already a layer with this ID');
```

Comment:
:+1: 

note to self: I'll need to pass a `arrayIndex` parameter https://github.com/mapbox/mapbox-gl-style-spec/blob/mb-pages/lib/validate/validate_layer.js#L26